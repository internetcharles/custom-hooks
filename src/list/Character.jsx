import React from 'react';
import PropTypes from 'prop-types';

export const Character = ({ name, image }) => (
  <div>
    <h1>{name}</h1>
    <img src={image} alt={name} />
  </div>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Character;
