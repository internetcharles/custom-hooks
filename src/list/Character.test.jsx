import React from 'react';
import { render } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  it('renders character', () => {
    const { asFragment } = render (<Character
      name="Phil"
      image="phil.jpg"
      occupation="Man"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
