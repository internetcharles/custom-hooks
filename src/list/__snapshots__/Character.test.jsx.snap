// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders character 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Phil
    </h1>
    <img
      alt="Phil"
      src="phil.jpg"
    />
  </div>
</DocumentFragment>
`;
